core/src/consensus/tendermint/mod.rs

```diff
@@ -455,8 +456,17 @@ impl Tendermint {
         let height = proposal.number() as Height;
         let view = consensus_view(proposal).expect("Imported block is already verified");
         if current_height == height && self.view.load(AtomicOrdering::SeqCst) == view {
-            *self.proposal.write() = Some(proposal.hash());
-            *self.proposal_parent.write() = *proposal.parent_hash();
+            {
+                let mut proposal_guard = self.proposal.write();
```

Comment:
How about lock `proposal_parent` here?

Reply:
Is there any reason to split `self.proposal` and `self.proposal_parent`? I don't think it would improve performance.

Reply:
I agree.
Let's merge two locks in the other commit.

Reply:
Okay.